src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/caseprogression/JudgeFinalOrderGenerator.java

```diff
@@ -0,0 +1,97 @@
+        //else docmosisTemplate = ASSISTED_ORDER_PDF;
+        DocmosisDocument docmosisDocument =
+            documentGeneratorService.generateDocmosisDocument(templateData, docmosisTemplate);
+        return documentManagementService.uploadDocument(
+            authorisation,
+            new PDF(
+                getFileName(caseData, docmosisTemplate),
+                docmosisDocument.getBytes(),
+                DocumentType.JUDGE_FINAL_ORDER
+            )
+        );
+    }
+
+    @Override
+    public JudgeFinalOrderForm getTemplateData(CaseData caseData) throws IOException {
+        return null;
+    }
+
+    private String getFileName(CaseData caseData, DocmosisTemplates docmosisTemplate) {
+        return String.format(docmosisTemplate.getDocumentTitle(), LocalDate.now());
+    }
+
+    private JudgeFinalOrderForm getFinalOrderType(CaseData caseData) {
+        return caseData.getFinalOrderSelection().equals(FREE_FORM_ORDER) ? getFreeFormOrder(caseData) : getAssistedOrder(
+            caseData);
+    }
+
+    private JudgeFinalOrderForm getFreeFormOrder(CaseData caseData) {
+        var freeFormOrderBuilder = JudgeFinalOrderForm.builder()
+            .caseNumber(caseData.getCcdCaseReference().toString())
+            .caseName(caseData.getCaseNameHmctsInternal())
+            .claimantReference(checkReference(caseData)
+                                   ? caseData.getSolicitorReferences().getApplicantSolicitor1Reference() : null)
+            .defendantReference(checkReference(caseData)
+                                   ? caseData.getSolicitorReferences().getRespondentSolicitor1Reference() : null)
+            .freeFormRecitalText(caseData.getFreeFormRecitalTextArea())
+            .freeFormRecordedText(caseData.getFreeFormRecordedTextArea())
+            .freeFormOrderedText(caseData.getFreeFormOrderedTextArea())
+            .orderOnCourtsList(caseData.getOrderOnCourtsList())
+            .onInitiativeSelectionText(nonNull(caseData.getOrderOnCourtInitiative())
+                                           ? caseData.getOrderOnCourtInitiative().getOnInitiativeSelectionTextArea() : null)
+            .onInitiativeSelectionDate(nonNull(caseData.getOrderOnCourtInitiative())
+                                           ? caseData.getOrderOnCourtInitiative().getOnInitiativeSelectionDate() : null)
+            .withoutNoticeSelectionText(nonNull(caseData.getOrderWithoutNotice())
+                                            ? caseData.getOrderWithoutNotice().getWithoutNoticeSelectionTextArea() : null)
+            .withoutNoticeSelectionDate(nonNull(caseData.getOrderWithoutNotice())
+                                            ? caseData.getOrderWithoutNotice().getWithoutNoticeSelectionDate() : null);
+        return freeFormOrderBuilder.build();
+    }
+
+    // will be added part of civ-7569
+    private JudgeFinalOrderForm getAssistedOrder(CaseData caseData) {
+        return null;
+    }
```

Comment:
I presume CIV-7569 will be worked on immediately afterwards?

Ideally I wouldn't want a placeholder implementation.